rig/machine_control/machine_controller.py

```diff
@@ -1352,7 +1352,21 @@ def __str__(self):
 
 
 class MemoryIO(object):
-    """A file-like view into a subspace of the memory-space of a chip."""
+    """A file-like view into a subspace of the memory-space of a chip.
+
+    A `MemoryIO` is subscriptable to allow construction of new, more specific,
+    file-like views of memory.
+
+    For example::
+
+        >>> f = MemoryIO(mc, 0, 1, 0x67800000, 0x6780000c)  # doctest: +SKIP
```

Comment:
Gah! stupid doctest! Comments I think... Aligned off to the right though, perhaps, to minimise visual noise?